### ISSUE TITLE
All wallets got deleted

### ISSUE THREAD

**Issue by maltokyo** [BlueWallet/BlueWallet#2529]
After upgrading to 6.0.3 (and using for a couple of days), opening it yesterday I was surprised to see no wallets at all! It was asking me to setup a brand new wallet. 

The only thing I changed on phone was that I turned on fingerprint authentication. After this the app no longer asked me for the passphrase that I had previously set, and asked for the fingerprint. After the first login with fingerprint, the deletion happened. 

I only had watch only wallets and one with a small amount (I have seed backup, so nothing lost), but I feel this is a critical bug somewhere, especially after seeing this user with the same issue: https://github.com/BlueWallet/BlueWallet/issues/2499

There is a bug, at least for me on Android and version 6.0.3, all my wallets got deleted, and I was asked to create a new one.

**Comment by ncoelho** [comment]
Yes, there was a known bug on iOS only code. Details and post mortem here https://github.com/BlueWallet/BlueWallet/wiki/Postmortem-3

But there's no reports on android. Can you reproduce the behaviour?

**Comment by maltokyo** [comment]
Is there a good testing tool or method I could use? Otherwise all I can do is try to make some wallets, and try to reproduce deletion, but even if I can, how can I share some log or something with you?

**Comment by ncoelho** [comment]
Being able to reproduce would be the first thing.

_update to 603 > turn on fingerprint on the app > stops asking fo passphrase > After the first login with fingerprint, the deletion happened._

This flow seems weird to me, the fingerprint does not or should not replace the passphrase.

**Comment by maltokyo** [comment]
Ok let me try to remember and reproduce that flow. In the meantime this definitely happened (multiple wallets deleted).. so please keep in mind.

**Comment by maltokyo** [comment]
I made a new wallet
Turned encryption on
Turned fingerprint on
And.. could not reproduce this. 

But, the funny thing is that initially I never turned fingerprint on in Bluewallet, only enabled it on the phone in Android settings.. 

The next time I logged in to Bluewallet I was asked for a fingerprint, which was super strange as I'd never used that before.  I was also NOT asked for my encryption password at all, which was strange too and rang alarm bells, and then of course I realised that it was completely empty inside. 

Nobody has access to this phone and I've never cleared data for the android app. 

Seems I can't help further, but confirm again that this issue 100% happened

**Comment by marcosrdz** [comment]
What Android version are you running?

**Comment by maltokyo** [comment]
This one

**Comment by marcosrdz** [comment]
Sorry, i mean what Android OS version are you running on your device, plus Make/Model?

**Comment by maltokyo** [comment]
Phone: Xiaomi Mi 9T
UI: MiUI 12 (all latest updates applied)
OS: Android 10 (all latest updates applied)

**Comment by slavik0329** [comment]
I lost over $500 with this issue

**Comment by josecitogarcia1963** [comment]
Same problem here. Today all my wallets are gone. I had $200 in a Lightning Wallet, and unfortunately I didn't find its backup. So, what should I do? Wait for a miracle update to get me access back to my Lighting Wallet?

I use Android (OnePlus 5T), Android 10.0.1

**Comment by ncoelho** [comment]
> Same problem here. Today all my wallets are gone. I had $200 in a Lightning Wallet, and unfortunately I didn't find its backup. So, what should I do? Wait for a miracle update to get me access back to my Lighting Wallet?
> 
> I use Android (OnePlus 5T), Android 10.0.1

Waiting won't help, the wallets are being deleted from storage, at least that seems clear from the symptoms described. If you have any invoice generated or refill transaction ID we can try to recover the Lightning wallet, if you do, please send an email to support at bluewallet@bluewallet.io

Please don't forget to backup your keys, any situation like this can happen at any time. Meanwhile we are going to keep investigating to see if we find any clue on what my be happen here.

**Comment by josecitogarcia1963** [comment]
> > Same problem here. Today all my wallets are gone. I had $200 in a Lightning Wallet, and unfortunately I didn't find its backup. So, what should I do? Wait for a miracle update to get me access back to my Lighting Wallet?
> > I use Android (OnePlus 5T), Android 10.0.1
> 
> Waiting won't help, the wallets are being deleted from storage, at least that seems clear from the symptoms described. If you have any invoice generated or refill transaction ID we can try to recover the Lightning wallet, if you do, please send an email to support at [bluewallet@bluewallet.io](mailto:bluewallet@bluewallet.io)
> 
> Please don't forget to backup your keys, any situation like this can happen at any time. Meanwhile we are going to keep investigating to see if we find any clue on what my be happen here.

Thanks for replying. I found my lightning backup (I knew it was somewhere). But I got a bad taste of mouth with the issue itself. So, I will wait for the resolution as soon as you have a conclusion (and fix) for this. Thanks

**Comment by ncoelho** [comment]
> Thanks for replying. I found my lightning backup (I knew it was somewhere). But I got a bad taste of mouth with the issue itself. So, I will wait for the resolution as soon as you have a conclusion (and fix) for this. Thanks

We don't have many clues on what is causing this, if it is a BlueWallet bug, a specific system behaviour (some phones reset the storage after some settings change), a user behaviour, etc. Only thing we have is two reports and on we know it is affecting android v10 apparently.

If you can give us more information, anything will really help. Like:
- Any android flavour or mod installed?
- Any non-default OS setting activated?
- Any encryption activated, on the app or OS level.
- What was the usage pattern, was the app on background or foreground?
- Can you reproduce the exact behaviour you did and get the same results?

Thanks for reporting, we do appreciate!

**Comment by maltokyo** [comment]
Is there some debug process we can turn on or something. To be honest I was also so shocked that it could happen that I loaded it now (and until the cause is found) with only read only wallets and wouldn't date to use lightning knowing this could happen).

**Comment by josecitogarcia1963** [comment]
> information

- Any android flavour or mod installed?
Android 10, OxygenOS 10.0.1 (OnePlus native OS)

- Any non-default OS setting activated?
Not that I can recall.

Any encryption activated, on the app or OS level.
I'm using a password to protect the app (I understand that this encrypts the Wallet, not sure though). Before the incident, whenever I opened the app, it asked me that password. Today, when the issue arose, It asked me the password and after that my fingerprint (which has never asked before as long as I can remember)

- What was the usage pattern, was the app on background or foreground?
Background.

- Can you reproduce the exact behaviour you did and get the same results?
The app has been working fine till today. As I explain, today I send some bitcoins to a wallet, which is imported as a watch-only wallet in BlueWallet.
I got the app notification for that transaction, which is crazy, because when I got into the app, everything was wiped out.
So, no wallets, but at the same time, the notification of the transaction was there. Which is awkward and misleading if you are saying that everything was probably removed from the device storage.

**Comment by ncoelho** [comment]
> It asked me the password and after that my fingerprint (which has never asked before as long as I can remember)

This is interesting, that tells us something already.

**Comment by maltokyo** [comment]
I had the same, exact same with the fingerprint! It only generally asked for the password only. I didn't add any settings related to fingerprint around that time and suddenly one day Bluewallet asked me for a fingerprint. Everything was deleted after I logged in, with this process. And it only happened just that one time.

**Comment by ncoelho** [comment]
Ok, can any of you try this please?

We suspect that Devices like oneplus and the other mentioned, wipe the keychain after changing device security settings, like the fingerprint authentication.

Su to reproduce it, 
- deactivate fingerprint auth on settings
- activate encryption on bluewallet with password
- Clear bluewallet from background
- activate fingerprint auth on settings
- Open bluewallet

Let me know...

**Comment by maltokyo** [comment]
I can do this. What does "Clear bluewallet from background" mean please. I'm not quite sure what I need to do. Clear all of its data?

**Comment by ncoelho** [comment]
> I can do this. What does "Clear bluewallet from background" mean please. I'm not quite sure what I need to do. Clear all of its data?

Just kill the app, so it asks fo password next time it is opened

**Comment by marcosrdz** [comment]
> I can do this. What does "Clear bluewallet from background" mean please. I'm not quite sure what I need to do. Clear all of its data?

Kill the app. Removing it from the multi task app switcher

**Comment by maltokyo** [comment]
Unfortunately (?) I just did all of this exactly as described, bit my wallets remain intact. Strange. I cannot make it happen now.
